packages/core/src/extension.ts

```diff
@@ -304,3 +304,32 @@ export async function emitUserState() {
         })
     })
 }
+
+// TODO: remove once version bump to 1.83.0 is complete.
+function setupVscodeVersionNotification() {
+    let notificationDisplayed = false
+    tryShowNotification()
+
+    function tryShowNotification() {
+        // Do not show the notification if the IDE version will continue to be supported.
+        if (!semver.gte(vscode.version, '1.83.0')) {
+            return
+        }
+
+        if (notificationDisplayed) {
+            return
+        }
+
+        notificationDisplayed = true
+
+        telemetry.toolkit_showNotification.emit({
```

Comment:
seems like we should have a `vscode.window.*` wrapper in `shared/utilities/messages.ts` that emits this (maybe optional by a boolean flag).